## 43% of end-end tests have passed
```
Total executed tests: 14
Failed tests: 5
Expected to fail tests: 3
Passed tests: 6
```
```
IDs of failed tests: 727231,741555,702745,741612,702843 
```
```
IDs of expected to fail tests: 741840,741839,741841 
```
<h3>Failed tests (5)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/rounding-send-bridge&PR_ID=22272&APK_NAME=22272.apk&TR_CASE_IDS=727231,741555,702745,741612,702843">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811717">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `accessibility id`: `share-qr-code-info-text` </div><div>Device 2: Tap on found: Text </div></blockquote></p>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:29: in prepare_devices
    self.public_key_2 = self.home_2.get_public_key()
../views/home_view.py:509: in get_public_key
    link_to_profile = self.get_link_to_profile()
../views/home_view.py:502: in get_link_to_profile
    link_to_profile = self.driver.get_clipboard_text()
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/extensions/clipboard.py:82: in get_clipboard_text
    return self.get_clipboard(ClipboardContentType.PLAINTEXT).decode('UTF-8')
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/extensions/clipboard.py:73: in get_clipboard
    base64_str = self.execute(Command.GET_CLIPBOARD, {'contentType': content_type})['value']
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: Cannot retrieve the current clipboard content from the device. Make sure the Appium Settings application is up to date. Original error: Error executing adbExec. Original error: 'Command '/root/lambda/shared-tools/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell ime set com.google.android.inputmethod.latin/com.android.inputmethod.latin.LatinIME' exited with code 1'; Command output: adb: device 'emulator-5554' not found
E   
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: Cannot retrieve the current clipboard content from the device. Make sure the Appium Settings application is up to date. Original error: Error executing adbExec. Original error: 'Command '/root/lambda/shared-tools/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell ime set com.google.android.inputmethod.latin/com.android.inputmethod.latin.LatinIME' exited with code 1'; Command output: adb: device 'emulator-5554' not found
E   
E       at getResponseForW3CError (/root/lambda/shared-tools/volume/appium/appium-2.1.3/node_modules/@appium/base-driver/lib/protocol/errors.js:1145:9)
E       at asyncHandler (/root/lambda/shared-tools/volume/appium/appium-2.1.3/node_modules/@appium/base-driver/lib/protocol/protocol.js:491:57)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESz3DpJPBOyrKC0q25">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESqrNMBnZlrZ4s41iu">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811722">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><p><blockquote><div># STEP: Adding new regular account</div><div>Device 1: Find `Button` by `accessibility id`: `add-account` </div></blockquote></p>

```
critical/wallet/test_wallet_mainnet.py:350: in test_wallet_add_remove_regular_account
    self.wallet_view.add_regular_account(account_name=new_account_name)
../views/wallet_view.py:245: in add_regular_account
    self.add_account_button.click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by accessibility id: `add-account` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811725">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td><p><blockquote><div># STEP: Checking the Swap flow for SNT on Optimism</div><div>Device 1: Find `Button` by `xpath`: `//*[@content-desc='token-network']/android.widget.TextView[@text='Status']` </div></blockquote></p>

```
critical/wallet/test_wallet_mainnet.py:163: in test_wallet_swap_flow_mainnet
    self.wallet_view.select_asset('Status')
../views/wallet_view.py:201: in select_asset
    xpath="//*[@content-desc='token-network']/android.widget.TextView[@text='%s']" % asset_name).click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//*[@content-desc='token-network']/android.widget.TextView[@text='Status']` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811726">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td><p><blockquote><div>Device 1: Could not reach Button by pressing system back button </div><div>Device 1: Find `Button` by `xpath`: `//android.view.ViewGroup[contains(@content-desc,'Account 1')]` </div></blockquote></p>

```
critical/wallet/test_wallet_mainnet.py:244: in test_wallet_bridge_flow_mainnet
    self.wallet_view.get_account_element().click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//android.view.ViewGroup[contains(@content-desc,'Account 1')]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811718">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><p><blockquote><div>Device 2: Looking for a message by text: Message AFTER edit 2 (Edited) </div><div>Device 2: Find `ChatElementByText` by `xpath`: `//*[starts-with(@text,'Message AFTER edit 2 (Edited)')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:383: in test_community_message_edit
    self.channel_2.set_reaction(message_text_after_edit)
../views/chat_view.py:690: in set_reaction
    self.chat_element_by_text(message).long_press_without_release()
../views/base_element.py:315: in long_press_without_release
    action.click_and_hold(self.find_element()).perform()
../views/chat_view.py:69: in find_element
    self.wait_for_visibility_of_element(20)
../views/base_element.py:138: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: ChatElementByText by xpath:`//*[starts-with(@text,'Message AFTER edit 2 (Edited)')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESH5hegRgHAnVnPESj">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESvOmesuDfHpyOh4TV">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (3)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811728">test_wallet_send_collectible</a>, id: 741840 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Find `CollectibleItemElement` by `xpath`: `//*[@content-desc='collectible-list-item']//*[contains(@text,'BVL')]/../..` </div></blockquote></p>

```
critical/wallet/test_collectibles.py:106: in test_wallet_send_collectible
    self.wallet_view.get_collectible_element('BVL').click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: CollectibleItemElement by xpath: `//*[@content-desc='collectible-list-item']//*[contains(@text,'BVL')]/../..` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811727">test_wallet_collectibles_balance</a>, id: 741839 </b></td></tr><tr><td><p><blockquote><div>Device 1: `Text` is `Double Spike` </div><div>Device 1: Click system back button </div></blockquote></p>

```
critical/wallet/test_collectibles.py:95: in test_wallet_collectibles_balance
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: BVL image doesn't match expected template
E    Device 1: BVL quantity 2 doesn't match expected 1
E    Device 1: BVL expanded image doesn't match expected template
E    Device 1: Glitch Punks image doesn't match expected template 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811729">test_wallet_collectible_send_from_expanded_info_view</a>, id: 741841 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `collectibles-tab` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/wallet/test_collectibles.py:153: in test_wallet_collectible_send_from_expanded_info_view
    self.wallet_view.get_collectible_element('Glitch Punks').wait_for_element().click()
../views/base_element.py:120: in wait_for_element
    raise TimeoutException(
 Device `1`: `CollectibleItemElement` by` xpath`: `//*[@content-desc='collectible-list-item']//*[contains(@text,'Glitch Punks')]/../..` is not found on the screen after wait_for_element 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (6)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811719">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESGpGmY34DDtC8AAcZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811716">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESGpGmY34DDtC8AAcZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811723">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811724">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811721">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3811720">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/common/utils.cljs
+++ b/src/status_im/contexts/wallet/common/utils.cljs
@@ -112,7 +112,8 @@
       zero-value?                    "0"
       (< token-units usd-cent-value) "0"
       :else                          (number/remove-trailing-zeroes
-                                      (.toFixed token-units standardized-decimals-count)))))
+                                      (number/format-decimal-fixed token-units
+                                                                   standardized-decimals-count)))))
 
 (defn add-token-symbol-to-crypto-balance
   [crypto-balance token-symbol]

--- a/src/utils/number.cljs
+++ b/src/utils/number.cljs
@@ -97,3 +97,18 @@
   [num decimal-count]
   (let [decimal-part (second (string/split (str num) #"\."))]
     (or (nil? decimal-part) (<= (count decimal-part) decimal-count))))
+
+(defn format-decimal-fixed
+  "Formats a number `n` to `decimal-places` without rounding.
+   - Ensures the exact number of decimal places (including trailing zeros).
+   - Does not round up or down, just trims excess decimals."
+  [n decimal-places]
+  (let [bn        (money/bignumber n)
+        scale     (money/bignumber (money/from-decimal decimal-places))
+        truncated (if (money/less-than bn 0)
+                    (Math/ceil (money/mul bn scale))
+                    (Math/floor (money/mul bn scale)))]
+    (-> truncated
+        (money/bignumber)
+        (money/div scale)
+        (money/to-fixed decimal-places))))

--- a/src/utils/number_test.cljs
+++ b/src/utils/number_test.cljs
@@ -88,3 +88,19 @@
 
     (is (true? (utils.number/valid-decimal-count? 1234567890.12 2)))
     (is (false? (utils.number/valid-decimal-count? 1234567890.12345 3)))))
+
+(deftest format-decimal-fixed-test
+  (testing "Format decimal numbers correctly without rounding"
+    (is (= "123.45" (utils.number/format-decimal-fixed "123.456" 2)))
+    (is (= "123.456" (utils.number/format-decimal-fixed "123.456" 3)))
+
+    (is (= "123.45600" (utils.number/format-decimal-fixed "123.456" 5)))
+    (is (= "123.000" (utils.number/format-decimal-fixed "123.000" 3)))
+
+    (is (= "123" (utils.number/format-decimal-fixed "123.999" 0)))
+
+    (is (= "-123.4" (utils.number/format-decimal-fixed "-123.456" 1)))
+    (is (= "-123.45" (utils.number/format-decimal-fixed "-123.456" 2)))
+
+    (is (= "0.0006" (utils.number/format-decimal-fixed "0.000650462672354754" 4)))
+    (is (= "999999999.999" (utils.number/format-decimal-fixed "999999999.999999" 3)))))

